components/scream/src/physics/shoc/shoc_functions_f90.cpp

```diff
@@ -425,8 +425,8 @@ void calc_shoc_vertflux_f(Int shcol, Int nlev, Int nlevi, Real *tkh_zi,
   static constexpr Int num_arrays = 4;
 
   Kokkos::Array<view_2d, num_arrays> temp_d;
-  Kokkos::Array<size_t, num_arrays> dim1_sizes     = {shcol,  shcol, shcol, shcol};
-  Kokkos::Array<size_t, num_arrays> dim2_sizes     = {nlevi,  nlevi, nlev,  nlevi};
+  Kokkos::Array<size_t, num_arrays> dim1_sizes     = {(size_t)shcol,  (size_t)shcol, (size_t)shcol, (size_t)shcol};
```

Comment:
Uhm, I know C-style casts are short and less visually heavy than C++ ones. But there's a trade-off: no compiler checks, and, being less visible, they may go unnoticed to the person maintaining the code (and are harder to grep for).

The option of using `static_cast<size_t>(..)` is not very appealing, since it'd made those lines very long. So I'm gonna go ahead and propose a more radical solution: can we do away with size_t and use int all around?

Reply:
Yeah, I had a feeling that C casts might generate controversy. :-) But I agree that your proposal of using int (or Int) instead of size_t.

Reply:
To be clear, I'm ok using whatever integer for local variable that live and die inside a small loop. E.g., looping over entries of a vector (though, if possible, I like range-for loops for those). I am more concerned about mixing size_t and int across functions and files.

For what it worth, I try to use unsigned _very sparingly_. On the topic, I somewhat follow [this section](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#arithmetic) of the Cpp Core Guidelines, in particular ES.100 and ES.106

Reply:
Makes sense! I have certainly experienced a lot of problems with mixing signed and unsigned integers, so moving to ints for stored data sounds best. Thanks for the reference.

Reply:
Another option is to change the host_to_device/device_to_host API so that it expects ints.

Reply:
> Another option is to change the host_to_device/device_to_host API so that it expects ints.

@bartgol , is this what you were suggesting by "do away with size_t and use int all around"?

Reply:
Basically, yes. We could achieve the same result by templating on the integer type. I would prefer this last option actually, since if we later change back to size_t, we don't have to re-change the functions signatures.